pennylane/tape/interfaces/autograd.py

```diff
@@ -107,16 +97,17 @@ def _update_trainable_params(self):
         self.trainable_params = trainable_params
         self._all_parameter_values = params
 
-    def get_parameters(self, trainable_only=True):  # pylint: disable=missing-function-docstring
-        self._update_trainable_params()
-        params = self._all_parameter_values
+    def get_parameters(self, trainable_only=True, return_arraybox=False):  # pylint: disable=missing-function-docstring
+        params = []
+        iterator = self.trainable_params if trainable_only else self._par_info
+
+        for p_idx in iterator:
+            op = self._par_info[p_idx]["op"]
+            op_idx = self._par_info[p_idx]["p_idx"]
+            params.append(op.data[op_idx])
 
-        if trainable_only:
-            params = [
-                p
-                for idx, p in enumerate(self._all_parameter_values)
-                if idx in self.trainable_params
-            ]
+        if return_arraybox:
+            return params

```

Comment:
In order to determine which parameters are trainable, we must _not_ wrap the parameters in an autograd list.